packages/components/src/date-time/date.js

```diff
@@ -1,135 +1,104 @@
-		if ( null === currentDate ) {
-			return null;
-		}
-		return currentDate ? moment( currentDate ) : moment();
-	}
-
-	// todo change reference to `isDayHighlighted` every time, `events` prop change
-	isDayHighlighted( date ) {
-		// Do not highlight when no events.
-		if ( ! this.props.events?.length ) {
-			return false;
-		}
-		if ( this.props.onMonthPreviewed ) {
-			this.props.onMonthPreviewed( date.toDate() );
-		}
+	return (
+		<div className="components-datetime__date--day-events">
+			<ul>
+				{ map( eventsToRender, ( event ) => (
+					<li>{ event.title || __( 'No title', 'gutenberg' ) }</li>
+				) ) }
+			</ul>
+		</div>
+	)
+};
 
-		// Compare date against highlighted events.
-		return this.props.events.some( ( highlighted ) =>
-			date.isSame( highlighted.date, 'day' )
-		);
+const renderDayContents = ( day, date, events ) => {
+	const eventsByDay = filter( events, ( ev ) => isSameDay( date, ev.date ) );
+	if ( ! eventsByDay?.length ) {
+		return getDate( date );
 	}
+	
+	return <>
+		<Tooltip text={ renderTooltipContent( eventsByDay ) }>
+			<span>{ getDate( date ).toString() }</span>
+		</Tooltip>
+	</>;
+};
 
-	render() {
-		const { currentDate, isInvalidDate } = this.props;
-		const momentDate = this.getMomentDate( currentDate );
-
-		return (
-			<div className="components-datetime__date" ref={ this.nodeRef }>
-				<DayPickerSingleDateController
-					date={ momentDate }
-					daySize={ 30 }
-					focused
-					hideKeyboardShortcutsPanel
-					// This is a hack to force the calendar to update on month or year change
-					// https://github.com/airbnb/react-dates/issues/240#issuecomment-361776665
-					key={ `datepicker-controller-${
-						momentDate ? momentDate.format( 'MM-YYYY' ) : 'null'
-					}` }
-					noBorder
-					numberOfMonths={ 1 }
-					onDateChange={ this.onChangeMoment }
-					transitionDuration={ 0 }
-					weekDayFormat="ddd"
-					isRTL={ isRTL() }
-					isOutsideRange={ ( date ) => {
-						return isInvalidDate && isInvalidDate( date.toDate() );
-					} }
-					isDayHighlighted={ this.isDayHighlighted }
-					onPrevMonthClick={ this.keepFocusInside }
-					onNextMonthClick={ this.keepFocusInside }
-				/>
-			</div>
-		);
-	}
-}
+const DatePicker = ( {
+	onChange,
+	onMonthChange,
+	currentDate,
+	isInvalidDate,
+	locale,
+	events,
+} ) => {
+	const selected = typeof currentDate === 'string' ? new Date( currentDate ) : currentDate;
+	const highlightDates = events?.length ? map( events, 'date' ) : [];
+	return (
+		<ReactDatePicker
+			calendarClassName={ 'components-datetime__date' }
+			selected={ selected }
+			onChange={ onChange }
+			inline
+			renderCustomHeader={ ( props ) =>
+				<DatePickerHeader { ...props } locale={ locale } />
+			}
+			renderDayContents={ ( ...props ) =>
+				renderDayContents( ...props, events )
+			}
+			useWeekdaysShort={ true }
+			locale={ locale }
+			highlightDates={ highlightDates }
+			onMonthChange={ onMonthChange }
+		/>
+	);
+};
 
 export default DatePicker;
+
```

Comment:
There's an extraneous newline here that should be removed. I'd recommend installing a [Prettier extension](https://prettier.io/docs/en/editors.html) and [EditorConfig extension](https://editorconfig.org/#download) for your IDE to automatically catch and fix formatting issues like this.